### PR TITLE
cmake: use mimalloc instead of nedmalloc

### PR DESCRIPTION
**Pull request by dscho**
Just like in the regular GCC (MINGW) build, let's do the same in the CMake (Visual C) build.

This is a long-overdue companion patch to https://github.com/git-for-windows/git/pull/4013.

### CODE REVIEW

**Comment by Murad1819** [comment]
// TensorFlow.js ni chaqirish
const tf = require('@tensorflow/tfjs');

// Modelni yaratish
const model = tf.sequential();
model.add(tf.layers.dense({units: 1, inputShape: [1]}));

// Modelni o'qitish uchun ma'lumotlar tayyorlash
const xs = tf.tensor2d([1, 2, 3, 4], [4, 1]);
const ys = tf.tensor2d([1, 3, 5, 7], [4, 1]);

// O'qitish jarayoni
model.compile({optimizer: 'sgd', loss: 'meanSquaredError'});
model.fit(xs, ys, {epochs: 10}).then(() => {
  // Test qilish
  const testInput = tf.tensor2d([5], [1, 1]);
  const prediction = model.predict(testInput);
  prediction.print();
});

**Comment by dscho** [comment]
Just like https://github.com/git-for-windows/git/pull/3685, I will close this PR because it is not worth spending any more time and love and energy on CMake support that is on its way out.
